extensions/wikia/Discussions/maintenance/ForumDumper.php

```diff
@@ -100,10 +102,13 @@ public function getPages() {
 
 		$dbh = wfGetDB( DB_SLAVE );
 		( new \WikiaSQL() )
-			->SELECT_ALL()
+			->SELECT( "page.*, comments_index.*, IF(pp.props is NULL, page.page_id, REPLACE(REPLACE(REPLACE(pp.props, ';', ''), 'i', ''), ':', '')) as idx" )
```

Comment:
Actually, why bother with this at all?  The values as-is sort correctly and aren't particularly special (in fact as multi-digit page IDs they're kind of gross).  Initialize a PHP var to zero below in the loop, increment it for each row and assign that. (you'll have to change the sort to ascending).